### PR TITLE
Extract VCloseButton and add descriptive labels

### LINKED ISSUE #609: Add descriptive aria-label to content report popover close button

### LINKED ISSUE #618: Add required `aria-label` prop to VIconButton
## Problem

The IconButton component should always have an aria- label unless it is `aria-hidden`, which it shouldn't probably ever be considering it's an interactive part of the page. Without the label screen readers don't have the necessary information to describe the button to the user.

## Description

We should add a required prop for the `aria-label` attribute to enforce that it is always passed when using this component.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #618 by @sarayourfriend 
Fixes #609 by @sarayourfriend 

## Description

This PR extracts the close button into a new component, `VCloseButton`.

Some labels are updated to be more descriptive:
- Translation Banner: "Close the translation contribution help request banner"
- Migration Notice: "Close the announcement"
- Content settings menu on mobile: "Close the content settings menu"
- Internal header pages menu black modal: "Close the pages menu"

To make the buttons work well with the `focus-` custom Tailwind styles, I made it possible to remove the border from the filled VIconButton.

## Testing Instructions

Check the popovers (external sources, license explanation, report form), modals (mobile content settings, pages menu). The buttons should look and behave the same, but the name that you hear when using Screen reader should be more descriptive.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/components/VHeader/VHeaderMobile/VContentSettingsModalContent.vue [hunk 1]
```diff
@@ -35,11 +35,10 @@
...
           :applied-filter-count="appliedFilterCount"
         />
-        <VIconButton
-          class="self-center ms-auto hover:bg-dark-charcoal hover:text-white"
-          :icon-props="{ iconPath: closeIcon }"
-          :aria-label="$t('modal.aria-close')"
-          @click="close"
+        <VCloseButton
+          :label="$t('modal.close-content-settings')"
+          class="self-center ms-auto"
```
Comment (obulat) [inline frontend/src/components/VHeader/VHeaderMobile/VContentSettingsModalContent.vue:41]:
```suggestion
          variant="filled-white"
          class="self-center ms-auto"
```

**Review by dhruvkb — CHANGES_REQUESTED** [review]
The `VCloseButton` component could benefit from stories and snapshots of those stories. It's a bit hard to picture the 5 variants based on solely names.

Also the fact that none of the existing stories changed is good (if it means that the button remains visually unchanged) or not-so-good (if is means that the button appears in none of them).

File: frontend/src/components/VBanner/VNotificationBanner.vue [hunk 2]
```diff
@@ -48,20 +41,13 @@ export default defineComponent({
       type: String as PropType<BannerId>,
       required: true,
     },
+    closeButtonLabel: {
+      type: String,
```
[the remark this answers is not in the document]
Comment (obulat) [reply frontend/src/components/VBanner/VNotificationBanner.vue:45]: Wouldn't we have to update all the places that expect a string and we pass it a TranslateResult? 
I think that the Vue 3 version of `i18n` returns a string, so we will be able to just remove all of the `toString()`s from the codebase once we upgrade.

[the remark this answers is not in the document]
Comment (obulat) [reply frontend/src/components/VBanner/VNotificationBanner.vue:45]: I didn't realize that your comment is only about this specific component, `VCloseButton`. I am going to update this :)

**Comment by obulat** [comment]
> Also the fact that none of the existing stories changed is good (if it means that the button remains visually unchanged) or not-so-good (if is means that the button appears in none of them).

The buttons did not appear in stories, but they _did_ appear in other snapshots (External sources, global audio player, filter and header modals), and I had to tweak some things to make them pass. One such change was adding `focus-visible:` modifier to the `after:` element in the `focus` utilities. Without this change, even though the button on the global audio player did not have a border set,  it had a ring around it in resting state.

Great idea about stories! I've added them, but did not add tests. Do you think it's worth bloating the snapshots directory and making the tests longer, if we already have the snapshots in the components or pages VR tests, @dhruvkb ?

File: frontend/src/components/meta/VCloseButton.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    return { args }
+  },
+  data: () => ({
+    buttonVariants,
+  }),
+})
+
+<Canvas>
+  <Story
+    name="VCloseButton varuabts"
```
Comment (dhruvkb) [inline frontend/src/components/meta/VCloseButton.stories.mdx:92]:
```suggestion
    name="VCloseButton variants"
```

**Review by dhruvkb — APPROVED** [review]
Thanks for adding the story, it made the review so much easier thanks to the preview. As for the tests, if they're captured in other existing snapshots, I don't see any reason for duplicating them.

LGTM, except for a typo, but I won't block on it.

**Review by zackkrida — APPROVED** [review]
@obulat in testing compared to production, I noticed the translation notification bar appears taller:

CleanShot 2023-04-10 at 09 55 20@2x
CleanShot 2023-04-10 at 09 54 58@2x

However, this may be related to a production issue with Arabic in particular, as I'm seeing much more English here than locally:

https://openverse.org/ar

I believe the PR is correct though and this is some unrelated production issue.

**Pull request merged**